pkg/cache/v3/simple.go

```diff
@@ -366,8 +389,61 @@ func createResponse(request *Request, resources map[string]types.ResourceWithTtl
 	}
 }
 
+// CreateDeltaWatch returns a watch for a delta xDS request which implements the Simple SnapshotCache.
 func (cache *snapshotCache) CreateDeltaWatch(request *DeltaRequest, st *stream.StreamState) (chan DeltaResponse, func()) {
-	return nil, nil
+	nodeID := cache.hash.ID(request.Node)
+	t := request.GetTypeUrl()
+
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	info, ok := cache.status[nodeID]
+	if !ok {
+		info = newStatusInfo(request.Node)
+		cache.status[nodeID] = info
+	}
+
+	// update last watch request times
+	info.SetLastDeltaWatchRequestTime(time.Now())
+
+	value := make(chan DeltaResponse, 1)
+
+	// find the current cache snapshot for the provided node
+	snapshot, exists := cache.snapshots[nodeID]
+
+	// if respondDelta returns nil, there is no change in any nested resource version against the previous snapshot
+	// if that is the case we want to trigger new watches
+	if !exists || (respondDelta(request, value, st, snapshot, cache.log) == nil) {
+		watchID := cache.nextDeltaWatchID()
+		if cache.log != nil {
+			cache.log.Infof("open delta watch ID:%d for %s Resources:%v from nodeID: %q, system version %q", watchID,
+				t, st.ResourceVersions, nodeID, snapshot.GetVersion(t))
+		}
+
+		info.SetDeltaResponseWatch(watchID, DeltaResponseWatch{Request: request, Response: value, StreamState: st})
+
+		return value, cache.cancelDeltaWatch(nodeID, watchID)
+	}
+
+	return value, nil
+}
+
+func (cache *snapshotCache) nextDeltaWatchID() int64 {
+	return atomic.AddInt64(&cache.deltaWatchCount, 1)
+}
+
+// cancellation function for cleaning stale delta watches
+func (cache *snapshotCache) cancelDeltaWatch(nodeID string, watchID int64) func() {
+	return func() {
+		// uses the cache mutex
```

Comment:
comment seems redundant

Reply:
Ah yup sorry I was supposed to pull that out already 😄